micro-rdk/src/common/data_manager.rs

```diff
@@ -383,6 +400,10 @@ where
                                 current_chunk_size = next_message.len();
                                 current_chunk = vec![next_message];
                             }
+                            #[cfg(feature = "esp32")]
+                            unsafe {
+                                data_upload_callback();
```

Comment:
Is there any metadata we want to provide to the callback? Size of the upload? Time upload started? Duration of upload? Is there any information we want to be able to have the callback provide back to us? Once we release this API we won't be able to change it easily without breaking ABI.

Reply:
I don't think so. It's here mainly for debugging data upload from a C-FFI side and not intended to be used in normal ops . We should consider whether a more general API make senses longer term but that's a scope doc. But to your point we can feature gate it with something like "unstable/data-upload-hook" and not enable it by default so its easier to yank it in the future